# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /home/ofer/proj/joomla/public
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: administrator/components/com_redirect/models/forms/link.xml
--- administrator/components/com_redirect/models/forms/link.xml Base (BASE)
+++ administrator/components/com_redirect/models/forms/link.xml Locally Modified (Based On LOCAL)
@@ -62,6 +62,23 @@
 		</field>
 
 		<field
+			name="redirect_type"
+			type="list"
+			label="COM_REDIRECT_TYPE"
+			description="COM_REDIRECT_TYPE_DESC"
+			class="inputbox"
+			size="1"
+			required="true"
+			default="303">
+			<option
+				value="301">
+				COM_REDIRECT_MOVED_PERMANENTLY</option>
+			<option
+				value="303">
+				COM_REDIRECT_MOVED_TEMPORARY</option>
+		</field>
+
+		<field
\ No newline at end of file
 			name="referer"
 			type="text"
 			id="referer"
Index: administrator/components/com_redirect/views/link/tmpl/edit.php
--- administrator/components/com_redirect/views/link/tmpl/edit.php Base (BASE)
+++ administrator/components/com_redirect/views/link/tmpl/edit.php Locally Modified (Based On LOCAL)
@@ -52,6 +52,12 @@
 				<?php echo $this->form->getInput('published'); ?></li>
 			</ul>
 
+			<legend><?php echo JText::_('COM_REDIRECT_TYPES'); ?></legend>
+			<ul class="adminformlist">
+				<li><?php echo $this->form->getLabel('redirect_type'); ?>
+				<?php echo $this->form->getInput('redirect_type'); ?></li>
+			</ul>
+
 		</fieldset>
 		<fieldset class="adminform">
 			<legend><?php echo JText::_('COM_REDIRECT_DETAILS'); ?></legend>
Index: administrator/language/en-GB/en-GB.com_redirect.ini
--- administrator/language/en-GB/en-GB.com_redirect.ini Base (BASE)
+++ administrator/language/en-GB/en-GB.com_redirect.ini Locally Modified (Based On LOCAL)
@@ -53,3 +53,8 @@
 COM_REDIRECT_SEARCH_LINKS="Search in link fields."
 COM_REDIRECT_XML_DESCRIPTION="This component implements link redirection"
 JLIB_RULES_SETTING_NOTES="1. If you change the setting, it will apply to this component. Note that:<br /><em>Inherited</em> means that the permissions from global configuration and parent group will be used.<br /><em>Denied</em> means that no matter what the global configuration or parent group settings are, the group being edited cannot take this action on this component.<br /><em>Allowed</em> means that the group being edited will be able to take this action for this component (but if this is in conflict with the global configuration or parent group it will have no impact; a conflict will be indicated by <em>Not Allowed (Locked)</em> under Calculated Settings).<br />2. If you select a new setting, click <em>Save</em> to refresh the calculated settings."
+COM_REDIRECT_TYPES="Redirect types"
+COM_REDIRECT_TYPE="Redirect type"
+COM_REDIRECT_TYPE_DESC="You can set the HTTP redirect type (temporary or permanently). See also: http://en.wikipedia.org/wiki/URL_redirection"
+COM_REDIRECT_MOVED_PERMANENTLY="Moved permanently"
+COM_REDIRECT_MOVED_TEMPORARY="Moved temporary"
\ No newline at end of file
Index: plugins/system/redirect/redirect.php
--- plugins/system/redirect/redirect.php Base (BASE)
+++ plugins/system/redirect/redirect.php Locally Modified (Based On LOCAL)
@@ -52,16 +52,22 @@
 			// See if the current url exists in the database as a redirect.
 			$db = JFactory::getDBO();
 				$db->setQuery(
-				'SELECT '.$db->quoteName('new_url').', '.$db->quoteName('published'). 
+				'SELECT '.$db->quoteName('new_url').', '.$db->quoteName('redirect_type').', '.$db->quoteName('published'). 
 				' FROM '.$db->quoteName('#__redirect_links') .
 				' WHERE '.$db->quoteName('old_url').' = '.$db->quote($current),
 				0, 1
 			);
 			$link = $db->loadObject();
 
-			// If a redirect exists and is published, permanently redirect.
+			// If a redirect exists and is published.
 			if ($link and ($link->published == 1)) {
-				$app->redirect($link->new_url, null, null, true, false);
+				// set the redirect type
+				if ($link->redirect_type==301) {
+					$moved = true; // permanently
+				} else {
+					$moved = false; // temporary
+				}
+				$app->redirect($link->new_url, null, null, true, $moved);
 			}
 			else
 			{
